src/AzureIoTHub.Portal.Infrastructure/Services/AWS/GreenGrassService.cs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) CGI France. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace AzureIoTHub.Portal.Infrastructure.Services.AWS
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Amazon.GreengrassV2;
+    using Amazon.GreengrassV2.Model;
+    using AutoMapper;
+    using AzureIoTHub.Portal.Application.Managers;
+    using AzureIoTHub.Portal.Application.Services;
+    using AzureIoTHub.Portal.Domain;
+    using AzureIoTHub.Portal.Domain.Entities;
+    using AzureIoTHub.Portal.Domain.Exceptions;
+    using AzureIoTHub.Portal.Domain.Repositories;
+    using AzureIoTHub.Portal.Models.v10;
+    using AzureIoTHub.Portal.Shared.Models.v10.Filters;
+    using Microsoft.AspNetCore.Http;
+
+    public class GreenGrassService : IEdgeModelService
+    {
+        private readonly IAmazonGreengrassV2 greengras;
+        private readonly IDeviceModelImageManager deviceModelImageManager;
+        private readonly IMapper mapper;
+        private readonly IEdgeDeviceModelRepository edgeModelRepository;
+        private readonly IUnitOfWork unitOfWork;
+
+        public GreenGrassService(
+            IAmazonGreengrassV2 greengras,
+            IDeviceModelImageManager deviceModelImageManager,
+            IMapper mapper,
+            IEdgeDeviceModelRepository edgeModelRepository,
+            IUnitOfWork unitOfWork)
+        {
+            this.greengras = greengras;
+            this.deviceModelImageManager = deviceModelImageManager;
+            this.mapper = mapper;
+            this.edgeModelRepository = edgeModelRepository;
+            this.unitOfWork = unitOfWork;
+        }
+
+        public async Task CreateGreenGrassDeployment(IoTEdgeModel edgeModel)
+        {
+            var createDeploymentRequest = new CreateDeploymentRequest
+            {
+                DeploymentName = edgeModel?.Name,
+                Components = await CreateGreenGrassComponents(edgeModel!),
+                TargetArn = "arn:aws:iot:eu-west-1:578920151383:thinggroup/test" //How?
```

Comment:
The thing group should be named after the edge model.
Here search the corresponding thing group inside the AWS IoT Core and provide its ARN